src/animated/Controller.ts

```diff
@@ -127,25 +100,33 @@ class Controller<DS extends object = any> {
       const queue = (this.prevQueue = this.queue)
       this.queue = prevQueue
 
-      // Reset delay/async tracking
+      // Reset any queue-related state
       this.pendingCount = 0
+      this.onEndQueue.length = 0
+
+      // Never assume that the last update always finishes last, since that's
+      // not true when 2+ async animations have indeterminate durations.
+      let remaining = queue.length
```

Comment:
The controller no longer assumes that the last "update" will always be the last to finish. 🎉